modules/environment/login/login-inner.nix

```diff
@@ -62,19 +59,9 @@ writeText "login-inner" ''
     fi
   ''}
 
-  [ "$#" -gt 0 ] || echo "Sourcing Nix environment..."
-  . $HOME/.nix-profile/etc/profile.d/nix.sh
-
-  if [ -e "$HOME/.nix-profile/etc/profile.d/hm-session-vars.sh" ]; then
-    [ "$#" -gt 0 ] || echo "Sourcing home-manager environment..."
-    export NIX_PATH=$HOME/.nix-defexpr/channels''${NIX_PATH:+:}$NIX_PATH
-    set +u
-    . "$HOME/.nix-profile/etc/profile.d/hm-session-vars.sh"
-    set -u
-  fi
-
-  # Workaround for https://github.com/NixOS/nix/issues/1865
-  export NIX_PATH=nixpkgs=$HOME/.nix-defexpr/channels/nixpkgs/:$NIX_PATH
+  set +u
```

Comment:
I think this should be moved inside the file.

Reply:
I think it does not belong in the file because the `set +u` and `set -u` only make sense if the script that sources the file has `set -u`, otherwise that could result in unexpected behaviour.

Reply:
True, and saving/restoring the options, while possible, isn't nice. Fine, no more comments then.